<launch>
  <include file="$(find jsk_fetch_startup)/jsk_fetch.machine" />

  <!-- add jsk startups -->
  <node pkg="jsk_fetch_startup" name="warning" type="warning.py" respawn="true" />
  <node pkg="jsk_fetch_startup" name="nav_speak" type="nav_speak.py" respawn="true" />
  <node pkg="jsk_fetch_startup" name="boot_sound" type="boot_sound.py" />

  <!-- Buffer Server -->
  <node pkg="tf2_ros" type="buffer_server" name="tf2_buffer_server" output="screen">
    <param name="buffer_size" value="120.0"/>
  </node>

  <!-- jsk_pr2_startup/jsk_pr2_lifelog/db_client.launch -->
  <!-- mongodb -->
  <param name="robot/name" value="fetch" />
  <include file="$(find jsk_robot_startup)/lifelog/mongodb.launch" >
    <arg name="use_daemon" value="true"/>
    <arg name="port" value="27017" />
    <arg name="repl_set_mode" value="false" />
    <arg name="replicate" default="false" />
  </include>
  <!-- logging and resoring pr2 position data -->
  <arg name="map_frame" value="map" />
  <node name="move_base_db"
        pkg="jsk_pr2_startup" type="move_base_db.py" >
    <param name="map_frame" value="$(arg map_frame)"/>
    <param name="robot_frame" value="base_link"/> <!-- fetch does not have base_footprint -->
    <param name="update_cycle" value="0.1" />
  </node>

  <!-- include fetch_navigation -->
  <include file="$(find fetch_navigation)/launch/fetch_nav.launch" >
    <arg name="map_file" default="$(find jsk_maps)/raw_maps/eng2-7f-0.05.yaml" />
    <arg name="map_keepout_file" default="$(find jsk_maps)/raw_maps/eng2-7f-0.05_keepout.yaml" />
    <arg name="use_keepout" default="true" />
  </include>
</launch>
